### ISSUE TITLE
Error fetching NFL data: module 'arrow' has no attribute 'FORMAT_W3C'

### ISSUE THREAD

**Issue by K-Money** [zacs/ha-nfl#23]
This error originated from a custom integration.

Logger: custom_components.nfl
Source: helpers/update_coordinator.py:182
Integration: NFL (documentation, issues)
First occurred: 8:10:58 PM (1 occurrences)
Last logged: 8:10:58 PM

Error fetching NFL data: module 'arrow' has no attribute 'FORMAT_W3C'

Home Assistant 2022.9.4 
Supervisor 2022.08.6 
Operating System 9.0 
Frontend 20220907.2 - latest

**Comment by zacs** [comment]
This is weird. I just checked and arrow has not deprecated the `FORMAT_W3C` function at all. 

Have you installed any new custom components lately? Maybe something that would keep arrow snapped to an old version that doesn't include `FORMAT_W3C`. Far fetched since it would need to be years old. 

Did the component work before? If so, did it stop working randomly, or after an upgrade or installing something else?

**Comment by K-Money** [comment]
I haven't installed any other components lately but this was my first time trying to install this nfl component. I tried searching the logs for a deeper stack trace but couldn't find anything beyond what I posted. Is there a way to tell what other components are using dependencies?

**Comment by K-Money** [comment]
I thought I'd try to manually install arrow to see if that could jimmy something loose. Not sure if I was successful, but even after a restart it didn't change anything and still gives the same error. [Screenshot of console](https://ibb.co/ry7GGHG)

**Comment by zacs** [comment]
Can you get into a Python prompt in the image and run this: `print(arrow.now().format(arrow.FORMAT_W3C))`

I suspect it will fail, although it should not!

**Comment by K-Money** [comment]
It looks like it works properly within python to give the date, but python is only accessible if I specifically use `python3`. Could that be related?
[Screenshot](https://ibb.co/gyrNh6P)

**Comment by zacs** [comment]
Hmmm yah I can't imagine why it would throw the error when typing it into the console work perfectly fine. It looks like that addon logs you in as root, and I'm sure HA runs as a different user. I wonder if you could add a terminal template function to check the output of the command...

Can you make a sensor like this:

```
sensor:
  - platform: command_line
    command: python3 -c "import arrow; print(arrow.now().format(arrow.FORMAT_W3C))"
    name: Arrow test
```

and see if it works?

**Comment by K-Money** [comment]
Created the sensor and it looks like the command does fail with a nonzero return value, but the return value of 1 doesn't seem super helpful.

Logger: homeassistant.components.command_line
Source: components/command_line/__init__.py:48
Integration: command_line ([documentation](https://www.home-assistant.io/integrations/command_line), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+command_line%22))
First occurred: 12:04:14 AM (1 occurrences)
Last logged: 12:04:14 AM
Command failed (with return code 1): python3 -c "import arrow; print(arrow.now().format(arrow.FORMAT_W3C))"

Also just to make sure to run a control test, I copied the same command string python3 -c ...etc into the ssh & terminal addon and it does execute correctly with the date/time as the output.

**Comment by zacs** [comment]
Bummer. The only two options I can think of are to get in as the `hassio` user and do a `pip install arrow` because the library clearly didn't install. Alternatively you could remove the customer componet, reboot, reinstall customer component, then reboot again. I'd hope it would install the dependency in this case.

**Comment by K-Money** [comment]
Success! Documenting for posterity.

I found that HassOS is actually running everything as root, so root is the right user to be under. Installing it under one of the other users didn't help. 

I tried modifying the manifest.json file to force a specific version of arrow. No dice.
I tried to force reinstall arrow with `--force-reinstall` and a particular version. No dice.

Using the info I found here in the [custom requirements section](https://developers.home-assistant.io/docs/creating_integration_manifest/#requirements). I ran `pip install arrow --target /config/deps`and then did a restart. That seemed to do the trick. It looks like the default `pip install` just puts it somewhere that Homeassistant wasn't looking for it.

No idea why the manifest requirement didn't do that itself, but that's for smarter folks than I to ponder.

I've rebooted a few times now and it seems that it's persistent. The arrow_test command_line sensor also works now as well.

Thanks for all the help troubleshooting this and for the component in the first place!
